src/com/amplitude/api/Diagnostics.java

```diff
@@ -73,20 +76,14 @@ Diagnostics setDiagnosticEventMaxCount(int diagnosticEventMaxCount) {
         this.diagnosticEventMaxCount = Math.max(diagnosticEventMaxCount, DIAGNOSTIC_EVENT_MIN_COUNT);
         this.diagnosticEventMaxCount = Math.min(this.diagnosticEventMaxCount, DIAGNOSTIC_EVENT_MAX_COUNT);
 
-        // resize unsent list and transfer any unsent events over
-        if (this.diagnosticEventMaxCount < unsentEvents.size()) {
-            for (int i = 0; i < unsentEvents.size() - this.diagnosticEventMaxCount; i++) {
-                unsentEvents.remove(0);
+        // check if need to downsize
+        if (this.diagnosticEventMaxCount < unsentErrorStrings.size()) {
+            for (int i = 0; i < unsentErrorStrings.size() - this.diagnosticEventMaxCount; i++) {
+                String errorString = unsentErrorStrings.remove(0);
```

Comment:
feels like this would throw a ConcurrentModificationException. Test carefully before merging, or change this to eliminate the possibility of that coming up